modules-load

```diff
@@ -1,19 +1,15 @@
 #!/bin/sh
 # modules-load [-n] [-v] - modules-load.d(5) compatible kernel module loader
 
-export PATH=/bin:/sbin
+export PATH=/usr/bin:/usr/sbin:/bin:/sbin
 
 {
 # Parameters passed as modules-load= or rd.modules-load= in kernel command line.
 sed -nr 's/,/\n/;s/(.* |^)(rd\.)?modules-load=([^ ]*).*/\3/p' /proc/cmdline
 
-# Find files /{etc,run,usr/lib}/modules-load.d/*.conf in that order.
-find -L /etc/modules-load.d /run/modules-load.d /usr/lib/modules-load.d \
-       	-maxdepth 1 -name '*.conf' -printf '%p %P\n' 2>/dev/null |
-# Load each basename only once.
-	sort -k2 -s | uniq -f1 | cut -d' ' -f1 |
-# Read the files, output all non-empty, non-comment lines.
-	tr '\012' '\0' | xargs -0 -r grep -h -v -e '^[#;]' -e '^$'
+# Read modules-load.d/* files, output all non-empty, non-comment lines.
+system-dirs modules-load.d | tr '\012' '\0' |
+	xargs -0 -r grep -h -v -e '^[#;]' -e '^$'
```

Comment:
grep too?

Reply:
I am not sure that the grep is avoidable, unless you would prefer sed? I could try to figure something else out, maybe a case statement while reading each line. But I quite like the pipeline here.

Reply:
I just dislike depending on so much stuff during early init, especially tools that tend to not work the same between busybox and coreutils.

Reply:
Well the point of this patch is that void-runit already uses these tools and flags successfully for modules loading. Any solution that applies to sysctl would also be useful there, and this patch will not exercise any tools or flags not already being used. If you want to avoid these flags, then that is a separate matter. I am just collapsing logic and implementation.